packages/checkbox/src/use-checkbox.ts

```diff
@@ -249,6 +249,18 @@ export function useCheckbox(props: UseCheckboxProps = {}) {
     }
   }, [inputRef.current])
 
+  /**
+   * HTMLFormElement.reset() should reset the checkbox state
+   */
+  useSafeLayoutEffect(() => {
+    if (!inputRef.current) return
+    if (inputRef.current.form) {
+      inputRef.current.form.onreset = () => {
+        setCheckedState(!!defaultChecked)
+      }
+    }
+  }, [inputRef.current])
```

Comment:
```suggestion
  }, [])
```

I don't think we need to add this to dependencies given that it's a `ref`.